### PR TITLE
Corrections to TF code of section 18.9 (naive bayes)

### PR DESCRIPTION
**Pull request by biswajitsahoo1111**
Range of pixel values of MNIST digits in TF is from 0-255. For this section pixel values need to be converted to either 0 or 1. In the old TF code this was not done. So there were unexpected results.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1776/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1776/

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 1]
```diff
@@ -69,6 +69,15 @@ mnist_test = torchvision.datasets.MNIST(
 #@tab tensorflow
 ((train_images, train_labels), (
     test_images, test_labels)) = tf.keras.datasets.mnist.load_data()
+
+# Original pixel values of MNIST range from 0-255 (as the digits are stored as uint8). For this section,
+# pixel values that are greater than 128 (in the original image) are converted to 1 and
+# values that are less than 128 are converted to 0. Why? See section 18.9.2 and 18.9.3.
+train_images = tf.floor(tf.constant(train_images / 128, dtype = tf.float32))
```
Comment (terrytangyuan) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:76]: Consistent code style here.

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 2]
```diff
@@ -87,7 +96,7 @@ image.shape, label
 ```{.python .input}
 #@tab tensorflow
 image, label = train_images[2], train_labels[2]
-image.shape, label
+image.shape, label.numpy()
```
Comment (terrytangyuan) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:99]: Why is this needed?

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 3]
```diff
@@ -110,7 +119,7 @@ label, type(label)
 
 ```{.python .input}
 #@tab tensorflow
-label, type(label)
+label.numpy(), label.dtype
```
Comment (terrytangyuan) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:122]: Why is this needed?

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md [hunk 4]
```diff
@@ -361,7 +370,6 @@ py
 ```{.python .input}
 #@tab tensorflow
 log_P_xy = tf.math.log(P_xy)
-# TODO: Look into why this returns infs
```
Comment (terrytangyuan) [inline chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:364]: Is this fixed?

**Review by terrytangyuan — CHANGES_REQUESTED** [review]

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md (hunk 1 above)
Comment (biswajitsahoo1111) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:76]: Will fix it.

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md (hunk 2 above)
Comment (biswajitsahoo1111) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:99]: `tf.keras.datasets` gives `numpy.ndarray`. But at the beginning, I have converted `numpy.ndarray` to tf.Tensor. So if I don't use `label.numpy` answer would contain `shape`, `dtype` etc. To to just get the value of the label, I have used `label.numpy()`.

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md (hunk 3 above)
Comment (biswajitsahoo1111) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:122]: For the same reason as above. We  can ignore `label.numpy()`. But `label.dtype` is necessary. Otherwise, `type(label)` gives `tf.blah.blah.blah.EagerTensor`.

File: chapter_appendix-mathematics-for-deep-learning/naive-bayes.md (hunk 4 above)
Comment (biswajitsahoo1111) [reply chapter_appendix-mathematics-for-deep-learning/naive-bayes.md:364]: Yes.

**Comment by mli** [comment]
Job d2l-en/PR-1776/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1776/

**Review by terrytangyuan — APPROVED** [review]
